code/src/java/pcgen/gui2/facade/CharacterFacadeImpl.java

```diff
@@ -2164,14 +2166,39 @@ void refreshLanguageList()
 				currBonusLangs.add(lang);
 			}
 		}
-		int bonusLangRemain = bonusLangMax - currBonusLangs.size();
+		int bonusLangRemain = 0;
+		if (!theCharacter.getRace().getDisplayName().equals("<none selected>"))
+		{
+			bonusLangRemain = bonusLangMax - currBonusLangs.size();
+		}
 		if (!allowBonusLangAfterFirst && !atFirstLvl)
 		{
 			bonusLangRemain = 0;
 		}
 		numBonusLang.set(bonusLangRemain);
 		if (bonusLangRemain > 0)
 		{
+
+			boolean containsAddBonus = false;
+			/* Check to see if the chooserList already contains an "add bonus" chooser*/
+			for (LanguageChooserFacade chooser : langChoosersList)
+			{
+				/* If we find one, break, as we don't need to add another.*/
+				if (chooser.getName().equals(LanguageBundle.getString("in_sumLangBonus")))
```

Comment:
Same here, checking the name against the translated string seems odd. Is there another way of doing this?

Reply:
So far as I can tell, there isn't another way that I have been able to find. Do you have any suggestions?

Reply:
welp. Even reading later code in the file does the same thing and I don't see a better way. Tried modifying the interface for LanguageChooserFacade but that doesn't give us much.